content/en/blog/2025/stability-proposal-announcement.md

```diff
@@ -0,0 +1,266 @@
+authors have with building, operating, or designing telemetry systems at planet
+scale.
+
+A specification with no implementation is not a useful thing for end users,
+though. Developers and operators approach telemetry through a variety of lenses;
+Some organizations have high standards for observability, with entire teams
+dedicated to building internal monitoring and instrumentation frameworks. Other
+organizations view observability and monitoring as a second or third order
+priority -- something that needs to happen, but not something that's
+incentivized. OpenTelemetry, as a specification, needs to serve all of these
+users and their use cases.
+
+To make OpenTelemetry useful, we need to provide an 'on-ramp' from existing
+methods and modes, existing tools and strategies, which means we need to provide
+implementations of not just the specification, but _applications_ of it as well.
+In practice, this means we need to distribute libraries to add OpenTelemetry
+instrumentation to existing HTTP servers and clients, or Collector receivers to
+scrape metrics from MySQL and translate them into OTLP.
+
+It should not come as a surprise that most of the value that our users, and
+sponsors, derive from OpenTelemetry is through these instrumentation libraries
+and Collector components. We attempt to communicate the relationship between the
+core SDKs and these libraries/components via the `contrib` repository pattern,
+but the practical reality is that many end users do not perceive this
+distinction.
+
+For maintainers and project leadership, our stability goals and the nature of
+`contrib` present a significant challenge. Users want stable, well-tested, and
+performant releases -- that _also_ perform the same function as commercial
+instrumentation agents.
+
+### What's Changing?
+
+There's three major changes that we'll be rolling out. Keep in mind the exact
+details of some of this may change, but these goals are more or less fixed.
+
+1. All components across all repositories should adhere to a consistent way of
+   communicating stability, through a metadata file, that can be discovered and
+   parsed in a programmatic way. The exact format will be defined through an
+   OTEP and incorporated into the specification. As part of this, we'll be
+   normalizing stability levels across components, including semantic
+   conventions by introducing alpha/beta stability to that effort.
+2. Stability requirements will be expanded to include more requirements around
+   documentation and where it's hosted, example code, performance benchmarks
+   (where applicable), implementation cookbooks, and other artifacts as
+   necessary.
+3. Stable components must only include other stable components by default. A new
+   global configuration option shall be introduced that allows adopters to
+   choose a desired minimum stability level, with a default value of
+   `[stable, rc, beta]`.
```

Comment:
We can leave that part for the OTEP
```suggestion
   choose a desired minimum stability level.
```

Reply:
Hm, I think we want to commit to 'beta' being the default stability in this piece since we're also proposing adding beta stability for semconv?

Reply:
Okay, fair

Reply:
I think picking specific names and meaning for stability level is best done the OTEP.  Best message here is "we'll have a default value that is both stable and includes components you need for a complete solution".